frontend/apps/reader/modules/readersearch.lua

```diff
@@ -79,6 +80,12 @@ SRELL_ERROR_CODES[110] = _("No preceding expression in repetition.")
 SRELL_ERROR_CODES[111] = _("Expression too complex, some hits will not be shown.")
 SRELL_ERROR_CODES[666] = _("Expression may lead to an extremely long search time.")
 
+function ReaderSearch:registerKeyEvents()
+    if Device:hasKeyboard() then
+        self.key_events.ShowFulltextSearchInput = { { "S" }, args = true }
```

Comment:
```suggestion
        self.key_events.ShowFulltextSearchInput = { { "S" }, args = true }
        self.key_events.ShowFulltextSearchInputRecent = { { "Shift", "S" }, event = "ShowFulltextSearchInput" }
```

Reply:
may be make them both use an event, with the arg being true/false, so visually they really look like we're calling the same event handler

Reply:
like that?

Reply:
like what? :)

```lua
self.key_events.ShowFulltextSearchInputLast =  {          { "S" }, event = "ShowFulltextSearchInput", args = false }
self.key_events.ShowFulltextSearchInputBlank = { { "Shift", "S" }, event = "ShowFulltextSearchInput", args = true }

```

Reply:
args = false isn't really necessary so to my mind it's distracting

Reply:
Why to have more options (empty search string) on NT devices than on touch devices?

Reply:
hasKeyboard isn't non-touch unless you mean something else?

Reply:
Why to have more options (empty search string) on devices with keyboard than on devices without it?

Reply:
What are the reasons to have empty search string?

Reply:
I guess because it's more complicated to erase the text field (when you want to make a new search unrelated to the previous search term) with the Kindle physical keyboard than it is for us with long-press on backspace in the virtual keyboard.

Reply:
We can add something similar
https://github.com/koreader/koreader/blob/5affb68233fda994150df1da3d0b573402f20842/frontend/ui/widget/inputtext.lua#L631-L633
for NT users.

Reply:
I think it came from your suggestion. So I think we might agree that we can just leave it out. ;-)

Reply:
I actually like the idea of not having to delete the previous search every time, I mean one has a keyboard so it's not complicated to type stuff

Reply:
It can be

`input = search_string or self.last_search_text or self.ui.doc_settings:readSetting("fulltext_search_last_search_text"),`

called with `args = ""`.

Reply:
but isn't that just the same thing?

Reply:
The same result, yes.